browser/src/main/java/org/netbeans/html/presenters/browser/SimpleServer.java

```diff
@@ -0,0 +1,706 @@
+            }
+            if (host != null) {
+                LOG.log(Level.FINER, "Host {0}:{1}", new Object[]{host, port});
+            }
+            return new ReqRes(h, key, url, context, host, port, header, method, body);
+        }
+    }
+
+    final class ReqRes {
+
+        private final SelectionKey delegate;
+        private final Handler h;
+        final String url;
+        final String hostName;
+        final int hostPort;
+        final Map<String, String> args;
+        final String header;
+        final String method;
+        final ByteBuffer body;
+        private final ByteArrayOutputStream os = new ByteArrayOutputStream();
+        final Writer writer = new OutputStreamWriter(os, StandardCharsets.UTF_8);
+        final Map<String, String> headers = new LinkedHashMap<>();
+        String contentType;
+        int status = 200;
+        boolean computed;
+        boolean suspended;
+
+        public ReqRes(
+                Handler h, SelectionKey delegate,
+                String url, Map<String, String> args, String host,
+                int port, String header, String method, ByteBuffer body
+        ) {
+            this.h = h;
+            this.delegate = delegate;
+            this.url = url;
+            this.hostName = host;
+            this.hostPort = port;
+            this.header = header;
+            this.args = args;
+            this.method = method;
+            this.body = body;
+        }
+
+        void updateOperations() {
+            if (body != null && body.remaining() > 0) {
+                delegate.interestOps(SelectionKey.OP_READ);
+            } else if (suspended) {
+                delegate.interestOps(0);
+            } else {
+                delegate.interestOps(SelectionKey.OP_WRITE);
+            }
+        }
+
+        public WriteReply handle(SocketChannel channel) throws IOException {
+            if (!computed) {
+                computed = true;
+                h.service(SimpleServer.this, this, this);
+            }
+            if (suspended) {
+                channel.write(ByteBuffer.allocate(0));
+                return null;
+            }
+
+            if (contentType == null) {
+                contentType = "content/unknown"; // NOI18N
+            }
+
+            ByteBuffer bb = ByteBuffer.allocate(8192);
+            ((Buffer) bb).clear();
+
+            LOG.log(Level.FINE, "Serving page request {0}", url); // NOI18N
+            ((Buffer) bb).clear();
+            bb.put(("HTTP/1.1 " + status + "\r\n").getBytes());
+            bb.put("Connection: close\r\n".getBytes());
+            bb.put("Server: Browser presenter\r\n".getBytes());
+            bb.put(date(null));
+            bb.put("\r\n".getBytes());
+            bb.put(("Content-Type: " + contentType + "\r\n").getBytes());
+            for (Map.Entry<String, String> entry : headers.entrySet()) {
+                bb.put((entry.getKey() + ":" + entry.getValue() + "\r\n").getBytes());
```

Comment:
`getBytes` without an encoding is almost everytime broken. I would use either `ISO-8859-1` or `US-ASCII`.

Reply:
OK, good idea. Used since d2d8501.